### PR TITLE
add freeze option to git cirrus build

### PR DESCRIPTION
**Pull request by jcounts**
Adds a freeze option to git cirrus build which builds with un-pinned
sub dependencies and then re-pins the updated items and updates the requirement files appropriately.

GitHub: https://github.com/cloudant/service_engineering/issues/141

### CODE REVIEW

File: src/cirrus/build.py [hunk 1]
```diff
@@ -99,6 +104,14 @@ def execute_build(opts):
 
     # we have custom build controls in the cirrus.conf
     reqs_name = build_params.get('requirements_file', 'requirements.txt')
+    activate_cmd = '. ./{0}/bin/activate'.format(config.venv_name())
```
Comment (shudgston) [inline src/cirrus/build.py]: Remove that `0`

File: src/cirrus/build.py [hunk 2]
```diff
@@ -233,5 +250,58 @@ def main():
         build_docs(make_opts=opts.docs)
 
 
+class Requirements:
+    """
+    Modify requirments files
+    """
+    def __init__(self, dep_filename, sub_dep_filename, activate_cmd):
```
Comment (shudgston) [inline src/cirrus/build.py:257]: Add some docstrings for the arguments here.
Also, explain what "direct" means.

File: src/cirrus/build.py (hunk 1 above)
Comment (jcounts) [reply src/cirrus/build.py]: 15cde11

File: src/cirrus/build.py (hunk 2 above)
Comment (jcounts) [reply src/cirrus/build.py:257]: 17675d5
1a292fb

**Review by shudgston — APPROVED** [review]

**Pull request merged**